querier/src/table/state_fuser.rs

```diff
@@ -0,0 +1,472 @@
+//! 2. **Catalog Query:** Query parquet files and tombstones from catalog. It is important that this happens AFTER the
+//!    ingester request. This will create [`ParquetFileWithMetadata`] and [`Tombstone`].
+//! 3. **Pruning:** Call [`filter_parquet_files`] and [`filter_tombstones`] to filter out files and tombstones that are
+//!    too new (i.e. were created after between step 1 and 2).
+
+use std::{
+    collections::{HashMap, HashSet},
+    ops::Deref,
+    sync::Arc,
+};
+
+use data_types2::{
+    ParquetFileWithMetadata, PartitionId, SequenceNumber, SequencerId, Tombstone, TombstoneId,
+};
+use snafu::Snafu;
+
+use crate::ingester::IngesterPartition;
+
+/// Information about an ingester partition.
+///
+/// This is mostly the same as [`IngesterPartition`] but allows easier mocking.
+pub trait IngesterPartitionInfo {
+    fn partition_id(&self) -> PartitionId;
+    fn sequencer_id(&self) -> SequencerId;
+    fn parquet_max_sequence_number(&self) -> Option<SequenceNumber>;
+    fn tombstone_max_sequence_number(&self) -> Option<SequenceNumber>;
+}
+
+impl IngesterPartitionInfo for Arc<IngesterPartition> {
+    fn partition_id(&self) -> PartitionId {
+        self.deref().partition_id()
+    }
+
+    fn sequencer_id(&self) -> SequencerId {
+        self.deref().sequencer_id()
+    }
+
+    fn parquet_max_sequence_number(&self) -> Option<SequenceNumber> {
+        self.deref().parquet_max_sequence_number()
+    }
+
+    fn tombstone_max_sequence_number(&self) -> Option<SequenceNumber> {
+        self.deref().tombstone_max_sequence_number()
+    }
+}
+
+/// Information about a parquet file.
+///
+/// This is mostly the same as [`ParquetFileWithMetadata`] but allows easier mocking.
+pub trait ParquetFileInfo {
+    fn partition_id(&self) -> PartitionId;
+    fn min_sequence_number(&self) -> SequenceNumber;
+    fn max_sequence_number(&self) -> SequenceNumber;
+}
+
+impl ParquetFileInfo for ParquetFileWithMetadata {
+    fn partition_id(&self) -> PartitionId {
+        self.partition_id
+    }
+
+    fn min_sequence_number(&self) -> SequenceNumber {
+        self.min_sequence_number
+    }
+
+    fn max_sequence_number(&self) -> SequenceNumber {
+        self.max_sequence_number
+    }
+}
+
+/// Information about a tombstone.
+///
+/// This is mostly the same as [`Tombstone`] but allows easier mocking.
+pub trait TombstoneInfo {
+    fn id(&self) -> TombstoneId;
+    fn sequencer_id(&self) -> SequencerId;
+    fn sequence_number(&self) -> SequenceNumber;
+}
+
+impl TombstoneInfo for Tombstone {
+    fn id(&self) -> TombstoneId {
+        self.id
+    }
+
+    fn sequencer_id(&self) -> SequencerId {
+        self.sequencer_id
+    }
+
+    fn sequence_number(&self) -> SequenceNumber {
+        self.sequence_number
+    }
+}
+
+#[derive(Snafu, Debug)]
+pub enum FilterParquetError {
+    #[snafu(display("Compactor processed file that the querier would need to split apart"))]
```

Comment:
```suggestion
    #[snafu(display("Compactor processed file that the querier would need to split apart which is not yet implemented"))]
```

I think it is helpful to include in this message that this is a known limitation (rather than an unexpected error. But maybe I am over thinking things 🤔